debug/raster-color.html

```diff
@@ -0,0 +1,308 @@
+        }
+        #gradientbg {
+            z-index: 1;
+            background: repeating-conic-gradient(#ccc 0% 25%, white 0% 50%) 50% / 20px 20px;
+        }
+        #gradient {
+            transition: all 0.3s ease-in-out;
+        }
+    </style>
+</head>
+
+<body>
+<div id="map"></div>
+<div id="controls"></div>
+<div id="gradientbg"><div id="gradient"></div></div>
+
+<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/dat-gui/0.7.7/dat.gui.min.js"></script>
+<script type="text/javascript" src="https://unpkg.com/d3@7.6.1/dist/d3.min.js"></script>
+<script src="../dist/mapbox-gl-dev.js"></script>
+<script src="../debug/access_token_generated.js"></script>
+<script>
+
+const presets = {
+    Satellite: {
+        tileset: "mapbox.satellite",
+        rasterColor: "Cubehelix",
+        rasterColorMix: "luminosity",
+        rasterColorRange: "[0, 1]"
+    },
+    Terrain: {
+        tileset: "mapbox.terrain-rgb",
+        rasterColor: "Hypsometric",
+        rasterColorMix: "decode terrain-rgb",
+        rasterColorRange: "[0, 8848]",
+        forcePngraw: true
+    },
+    "Population log-density": {
+        tileset: "rreusser.796swou6 (pop log-density)",
+        rasterColor: "PopDensity",
+        rasterColorMix: "decode population log-density",
+        rasterColorRange: "[-3, 4]",
+        forbid2x: true,
+        forcePngraw: true
+    },
+    'Clouds (GMGSI LWIR)': {
+        tileset: "rreusser.dv2uv1dw (GMGSI clouds)",
+        rasterColor: "Transparent white",
+        rasterColorMix: "luminosity",
+        rasterColorRange: "[0, 1]"
+    },
+    'Radar composite reflectivity (GFS)': {
+        tileset: "rreusser.b3c8wj9i (GFS refc)",
+        rasterColor: "Radar",
+        rasterColorMix: "luminosity",
+        rasterColorRange: "[0, 1]"
+    },
+    'Radar composite reflectivity (HRRR)': {
+        tileset: "rreusser.5vzfcdjn (HRRR refc)",
+        rasterColor: "Radar",
+        rasterColorMix: "luminosity",
+        rasterColorRange: "[0, 1]"
+    },
+};
+
+function GUIParams () {
+    this.preset = "Satellite";
+    this.opacity = 1;
+    Object.assign(this, presets[this.preset]);
+}
+
+const guiParams = new GUIParams();
+
+const mixes = {
+    luminosity: [ 0.2126, 0.7152, 0.0722, 0 ],
+    average: [1/3, 1/3, 1/3, 0],
+    red: [1, 0, 0, 0],
+    green: [0, 1, 0, 0],
+    blue: [0, 0, 1, 0],
+    "decode terrain-rgb": [
+        256 * 256 * 256 * 0.1,
+        256 * 256 * 0.1,
+        256 * 0.1,
+        -10000
+    ],
+    "decode population log-density": [
+        256 * 256 * 256 * 8.348993603394451e-07,
+        256 * 256 * 8.348993603394451e-07,
+        256 * 8.348993603394451e-07,
+        -9
+    ],
+}
+
+const ranges = {
+    "[0, 1]": [0, 1],
+    "[0, 8848]": [0, 8848],
+    "[-3, 4]": [-3, 4],
+};
+
+const tilesets = {
+    "mapbox.terrain-rgb": "mapbox.terrain-rgb",
+    "mapbox.satellite": "mapbox.satellite",
+    "rreusser.796swou6 (pop log-density)": "rreusser.796swou6",
```

Comment:
Included the sake of illustration and debugging, but we should probably remove tilesets on my account from the final PR.